src/backend/hip/subconfigure.m4

```diff
@@ -121,25 +95,28 @@ if test "${have_hip}" = "yes" ; then
                           ],)],[hip_version=${maj_version}],[])
         if test ! -z ${hip_version} ; then break ; fi
     done
-    PAC_PUSH_FLAG([IFS])
-    IFS=","
-    HIP_SM=
-    for sm in ${with_hip_sm} ; do
-          PAC_APPEND_FLAG([$sm],[HIP_SM])
-    done
-    PAC_POP_FLAG([IFS])
+    HIP_GENCODE=
+    if test "${with_hip_sm}" != "auto"; then
+        PAC_PUSH_FLAG([IFS])
+        IFS=","
+        HIP_SM=
+        for sm in ${with_hip_sm} ; do
+            PAC_APPEND_FLAG([$sm], [HIP_SM])
+        done
+        PAC_POP_FLAG([IFS])
 
-    for sm in ${HIP_SM} ; do
+        for sm in ${HIP_SM} ; do
+            if test -z "${HIP_GENCODE}" ; then
+                HIP_GENCODE="--offload-arch=${sm}"
+            else
+                HIP_GENCODE="${HIP_GENCODE} --offload-arch=${sm}"
+            fi
+        done
         if test -z "${HIP_GENCODE}" ; then
-            HIP_GENCODE="--offload-arch=${sm}"
-        else
-            HIP_GENCODE="${HIP_GENCODE} --offload-arch=${sm}"
+            AC_MSG_ERROR([--with-hip-sm not specified; either specify it or disable hip support])
```

Comment:
I am thinking that the default is `auto`.  If the user do `--with-hip-sm=xxx`, we directly pass to `--offload-arch=xxx`.

Reply:
Yeah, the message here is obsolete. We receive the option as comma-separated list, but HIPCC takes them as separate `--offload-arch=` options.